actix-router/src/quoter.rs

```diff
@@ -135,89 +92,59 @@ impl Quoter {
-/// - `0x30 ('0') => 0x0`
-/// - `0x39 ('9') => 0x9`
-/// - `0x41 ('a') => 0xA`
-/// - `0x61 ('A') => 0xA`
-/// - `0x46 ('f') => 0xF`
-/// - `0x66 ('F') => 0xF`
-fn from_ascii_hex(v: u8) -> Option<u8> {
-    match v {
-        b'0'..=b'9' => Some(v - 0x30),      // ord('0') == 0x30
-        b'A'..=b'F' => Some(v - 0x41 + 10), // ord('A') == 0x41
-        b'a'..=b'f' => Some(v - 0x61 + 10), // ord('a') == 0x61
-        _ => None,
-    }
-}
-
 /// Decode a ASCII hex-encoded pair to an integer.
 ///
 /// Returns `None` if either portion of the decoded pair does not evaluate to a valid hex value.
 ///
 /// - `0x33 ('3'), 0x30 ('0') => 0x30 ('0')`
 /// - `0x34 ('4'), 0x31 ('1') => 0x41 ('A')`
 /// - `0x36 ('6'), 0x31 ('1') => 0x61 ('a')`
+#[inline(always)]
 fn hex_pair_to_char(d1: u8, d2: u8) -> Option<u8> {
-    let (d_high, d_low) = (from_ascii_hex(d1)?, from_ascii_hex(d2)?);
+    let d_high = char::from(d1).to_digit(16)?;
+    let d_low = char::from(d2).to_digit(16)?;
 
     // left shift high nibble by 4 bits
-    Some(d_high << 4 | d_low)
+    Some((d_high as u8) << 4 | (d_low as u8))
 }
 
-/// Sets bit in given bit-map to 1=true.
-///
-/// # Panics
-/// Panics if `ch` index is out of bounds.
-fn set_bit(array: &mut [u8], ch: u8) {
-    array[(ch >> 3) as usize] |= 0b1 << (ch & 0b111)
+#[derive(Debug, Default, Clone)]
+struct AsciiBitmap {
+    array: [u8; 16],
 }
 
-/// Returns true if bit to true in given bit-map.
-///
-/// # Panics
-/// Panics if `ch` index is out of bounds.
-fn bit_at(array: &[u8], ch: u8) -> bool {
-    array[(ch >> 3) as usize] & (0b1 << (ch & 0b111)) != 0
+impl AsciiBitmap {
+    /// Sets bit in given bit-map to 1=true.
+    ///
+    /// # Panics
+    /// Panics if `ch` index is out of bounds.
+    fn set_bit(&mut self, ch: u8) {
+        self.array[(ch >> 3) as usize] |= 0b1 << (ch & 0b111)
+    }
+
+    /// Returns true if bit to true in given bit-map.
+    ///
+    /// # Panics
+    /// Panics if `ch` index is out of bounds.
+    fn bit_at(&self, ch: u8) -> bool {
+        self.array[(ch >> 3) as usize] & (0b1 << (ch & 0b111)) != 0
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    #[test]
-    fn hex_encoding() {
-        let hex = b"0123456789abcdefABCDEF";
-
-        for i in 0..256 {
-            let c = i as u8;
-            if hex.contains(&c) {
-                assert!(from_ascii_hex(c).is_some())
-            } else {
-                assert!(from_ascii_hex(c).is_none())
-            }
-        }
-
-        let expected = [
-            0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 10, 11, 12, 13, 14, 15,
-        ];
-        for i in 0..hex.len() {
-            assert_eq!(from_ascii_hex(hex[i]).unwrap(), expected[i]);
-        }
-    }
-
     #[test]
     fn custom_quoter() {
         let q = Quoter::new(b"", b"+");
         assert_eq!(q.requote(b"/a%25c").unwrap(), b"/a%c");
-        assert_eq!(q.requote(b"/a%2Bc").unwrap(), b"/a%2Bc");
+        assert_eq!(q.requote(b"/a%2Bc"), None);
```

Comment:
its definitely surprising that this returned something before, havent dug into why but definitely feels more like a fix than a break